src/elm/Questionnaires/Migration/Update.elm

```diff
@@ -0,0 +1,1158 @@
+            , getResultCmd result
+            )
+
+        ( _, _ ) ->
+            ( model, Cmd.none )
+
+
+
+-- ----------------
+-- Diff Root Node
+-- ----------------
+
+
+createDiffTree : KnowledgeModel -> FormValues -> Dict.Dict String TreeNode
+createDiffTree knowledgeModel replies =
+    let
+        children =
+            List.map .uuid knowledgeModel.chapters
+
+        path =
+            []
+
+        subtreeDict =
+            List.foldl (createChapterDiffSubTree [] replies) Dict.empty knowledgeModel.chapters
+
+        knowledgeModelNode =
+            { uuid = knowledgeModel.uuid
+            , treeUuid = knowledgeModel.uuid
+            , children = children
+            , title = knowledgeModel.name
+            , expansionState = Expanded
+            , path = path
+            , nodeType = KnowledgeModelType
+            }
+    in
+    Dict.insert knowledgeModel.uuid knowledgeModelNode subtreeDict
+
+
+createChapterDiffSubTree : Path -> FormValues -> Chapter -> Dict.Dict String TreeNode -> Dict.Dict String TreeNode
+createChapterDiffSubTree parentPath replies chapter diffTree =
+    let
+        children =
+            List.map getQuestionUuid chapter.questions
+
+        path =
+            List.append parentPath [ ChapterPathNode chapter.uuid ]
+
+        subtreeDict =
+            List.foldl (createQuestionDiffSubTree path replies) diffTree chapter.questions
+
+        chapterNode =
+            { uuid = chapter.uuid
+            , treeUuid = chapter.uuid
+            , children = children
+            , title = chapter.title
+            , expansionState = Collapsed
+            , path = path
+            , nodeType = ChapterType
+            }
+    in
+    Dict.insert chapter.uuid chapterNode subtreeDict
+
+
+createQuestionDiffSubTree : Path -> FormValues -> Question -> Dict.Dict String TreeNode -> Dict.Dict String TreeNode
+createQuestionDiffSubTree path replies question diffTree =
+    createQuestionDiffSubTreeAtKey path (getQuestionUuid question) replies question diffTree
+
+
+createQuestionDiffSubTreeAtKey : Path -> String -> FormValues -> Question -> Dict.Dict String TreeNode -> Dict.Dict String TreeNode
+createQuestionDiffSubTreeAtKey parentPath insertionKey replies question diffTree =
+    let
+        uuid =
+            getQuestionUuid question
+
+        path =
+            List.append parentPath [ QuestionPathNode insertionKey ]
+
+        replyPath =
+            stringifyPath path
+
+        answersChildren =
+            List.map .uuid <| getQuestionAnswers question
+
+        reply =
+            findQuestionReply replies replyPath
+
+        questionsChildren : List ( Int, List Question )
+        questionsChildren =
+            if List.length (getQuestionItemQuestions question) == 0 then
+                []
+
+            else
+                let
+                    -- We always want to see at least one item (even though it's not answered)
```

Comment:
I believe this is no longer true. Now, when there are no items in the questionnaire we only display the Add button. Is it somehow different in migrations?

Reply:
This is part of the questionnaire structure overview (the left panel). If the user answered by creating at leas one item, there will be a meta-note (marked as "Item 1", "Item 2" etc.) and if there is no answer, the questions from template will be shown directly under the template (see the attachment).
<img width="456" alt="diff tree" src="https://user-images.githubusercontent.com/5204614/58048482-211fb300-7b4b-11e9-81db-ce83bb288359.png">